draft-ietf-ppm-dap.md

```diff
@@ -1479,36 +1487,58 @@ struct {
 } InputShareAad;
 ~~~
 
+~~~ tls-presentation
+struct {
+ReportID id;
+ReportError error;
+} ReportUploadStatus;
+
+struct {
+ReportUploadStatus status<0..2^32-1>;
+} UploadResponse;
+~~~
+
 If the upload request is malformed, the Leader aborts with error
 `invalidMessage`.
 
 If the Leader does not recognize the task ID, then it aborts with error
 `unrecognizedTask`.
 
+Otherwise, the Leader responds with HTTP status 200, a body consisting
+of `UploadResponse` and the media type `application/dap-report-resp`. If
+the entire upload has been successfully completed, the Leader MAY respond with
+an empty body.
```

Comment:
I'm not in favor of this MAY. It's better I think if there's only one way to handle the response: expect an `UploadResponse`.

Reply:
I don't agree. We use empty response bodies in the aggregation and collection interactions to indicate that aggregation, aggregate share and collection jobs are being handled asynchronously, so using an empty body here seems fine, and saves a few bytes on the wire. The encoding of an empty upload response is bigger than nothing.

Reply:
Well, there's an important difference I think.

During aggregation, an empty response indicates async handling, which means the Leader needs to poll later. But there is no async processing of the upload request. Should there be?

Reply:
Sure, it's not exactly the same, but the point is that we already assign semantic meaning to the absence of a body. Further, the real issue here is that there's no point in constructing and parsing a message that conveys no information (the message would just be 4 0 values bytes indicating an empty vector of reports) when literally nothing would serve just as well.

Reply:
What I think is odd about this is that we have to special case the empty body. That is the code is something like:

```rust
fn parse_request_body(bytes: &[u8]) -> Vec<ReportStatus> {
   if bytes.empty() {
      return Vec::new();
   }
   parse_report_status_list(bytes)
}
```

But tbh I don't feel very strongly about this. I'm ok with whatever we we do here.